src/datumaro/plugins/sam_transforms/bbox_to_inst_mask.py

```diff
@@ -90,26 +92,35 @@ def __init__(
             launcher=self._sam_encoder_launcher,
             batch_size=1,
             append_annotation=False,
+            num_workers=num_workers,
         )
         self._to_polygon = to_polygon
 
-    def __iter__(self) -> Generator[DatasetItem, None, None]:
-        for batch in take_by(self._extractor, self._batch_size):
-            batch = [item for item in batch if self._launcher.type_check(item)]
-            img_embeds = self._sam_encoder_launcher.launch(batch)
+    def _process_batch(
+        self,
+        batch: List[DatasetItem],
+    ) -> List[DatasetItem]:
+        print("Process batch")
+        img_embeds = self._sam_encoder_launcher.launch(
+            batch=[item for item in batch if self._sam_encoder_launcher.type_check(item)]
+        )
 
-            for item, img_embed in zip(batch, img_embeds):
-                # Nested list of mask [[mask_0, ...]]
-                nested_masks: List[List[Mask]] = self._sam_decoder_launcher.launch(
-                    [item.wrap(annotations=item.annotations + img_embed)],
-                    stack=False,
-                )
+        items = []
+        for item, img_embed in zip(batch, img_embeds):
+            # Nested list of mask [[mask_0, ...]]
+            nested_masks: List[List[Mask]] = self._sam_decoder_launcher.launch(
+                [item.wrap(annotations=item.annotations + img_embed)],
+                stack=False,
+            )
 
-                yield item.wrap(
+            items.append(
+                item.wrap(
                     annotations=self._convert_to_polygon(nested_masks[0])
                     if self._to_polygon
                     else nested_masks[0]
                 )
+            )
```

Comment:
I think there are some unnecessary item wrapping.
```
batch_to_decode = [item.wrap(annotations=item.annotations + img_embed) for item, img_embed in zip(batch, img_embeds)]

nested_masks = self._sam_decoder_launcher.launch(batch_to_decode, stack=False)

items = []
for item, nested_mask in zip(batch, nested_masks):
    annotations = (self._convert_to_polygon(nested_masks[0]) if self._to_polygon else nested_masks[0])
    items.append(item.wrap(annotations=annotations))
```
I think it would be better to do item wrapping only once.

Reply:
https://github.com/openvinotoolkit/datumaro/pull/1145/commits/865fb471695a9635ddd37b76383e1adc746aac1c
decreases the number of calling `item.wrap` from two to one.